### DISCUSSION TITLE
Custom URI Budibase APP

### DISCUSSION THREAD

**Discussion by hblab-hautp** [Budibase/budibase#3811] [Help] [answered]
Hello everyone, 
After i published my app by budibase internal tool by using docker, I get an uri like that
http://localhost:8085/app_dev_1f33dfc4540f45868016678dfcffe95b#/budibasetest

Oh, it is not so pretty ! So, How can I remove or custom "non-beauty" string (i guess it is app key or app id) on my uri 
From : http://localhost:8085/app_dev_1f33dfc4540f45868016678dfcffe95b#/budibasetest
To : http://localhost:8085/budibasetest

Thank you very much ! Have a good day

**Comment by mike12345567** [comment, marked as answer]
Hi @hblab-hautp - there is an option for self hosted (not available on Cloud platform) to use the apps name for the URL. So for example if I have an app called `Sales` then I can access it via `http://localhost:8085/app/sales`. You'll need the `/app/` before the app name to tell Budibase that you want to open an app - also please note that spaces etc need to be URI encoded correctly so an app called `sales app` would be `/app/sales%20app`.

If you'd like to go further than this and use a subdomain or a specific path for an app then you'll need to setup a reverse proxy with something like NGINX or Caddy to that specific app URL - we have a guide about this in our docs [found here](https://docs.budibase.com/self-hosting/hosting-settings/advanced-options/reverse-proxy).

**Reply by hblab-hautp** [reply]
oh, thank you very much. I think now i can do it
But I didn't find this feature in your document.
`So for example if I have an app called Sales then I can access it via http://localhost:8085/app/sales`

**Reply by ryzam** [reply]
hi @mike12345567 
I try to access my app as your suggestion /app/myinternalapp, i can run curl and get the response but when try to browse its show this error

![image](https://user-images.githubusercontent.com/597759/147349296-b979a916-a2a0-4009-a01a-0d48681ceca1.png)
![image](https://user-images.githubusercontent.com/597759/147349399-ef8eb883-ae1d-4186-9532-e8a7e627f907.png)

**Reply by mike12345567** [reply]
Hi @ryzam - just to confirm, you don't need to setup any sort of nginx configuration for this to work in self host, it will work out of box. If you connect directly to Budibase (not through your NGINX proxy) it should automatically redirect based on your app name, e.g. If the app is called `myinternalapp` (if it contains spaces these need URI encoded) then the URL `/app/myinternalapp` will work.

**Reply by ryzam** [reply]
But then how to map our internal ip that server budibase to our domain url if not using nginx as reverse-proxy?Sorry I'm a bit confuse.

BTW i have this configuration to access builder page and it's working as expected and i also can access the apps
![image](https://user-images.githubusercontent.com/597759/147354358-4a2995e2-ea43-425b-8818-7c188d5d73d8.png)

![image](https://user-images.githubusercontent.com/597759/147354210-3072bdc2-091c-4bad-8fff-0d1d0c92a1a6.png)

![image](https://user-images.githubusercontent.com/597759/147354269-71af2741-35f2-4b20-b92f-941c7b0d3895.png)

**Reply by ryzam** [reply]
Update!
Ok i can access /app/jobs but not the public page. I need to sign-in.

![image](https://user-images.githubusercontent.com/597759/147354780-b13d3aad-6873-4f42-b937-88b4cb9b5d40.png)

**Reply by mike12345567** [reply]
Hi @ryzam yes you'll need the second config you provided to proxy from a sub domain on your own domain, but you won't need any per app config.

**Reply by ryzam** [reply]
Thank @mike12345567 , but I'm still facing problem to access apps with public role and if I try to connect using this url /app/jobs.  The page show blank when access public page as shown in the below image. Why it try redirect to builder path?

![image](https://user-images.githubusercontent.com/597759/147377384-5c676140-8f27-43e7-b2df-4223d7c65265.png)
